plugins/postgres/dbt/include/postgres/macros/adapters.sql

```diff
@@ -155,6 +155,22 @@
                                   })) -%}
 {% endmacro %}
 
+{% macro postgres__make_backup_relation(base_relation, suffix) %}
```

Comment:
Using your feedback as a jumping off point, this is what came together:

- `make_temp_relation` is used in snapshots and incremental materializations requiring the relation returned to not have `schema` or `database` defined, specifically on Postgres, to avoid `cannot create temporary relation in non-temporary schema`. In order to accomplish [your other suggestions](https://github.com/dbt-labs/dbt-core/pull/4921#discussion_r832994173) in view and table I created a `make_intermediate_relation` macro that returns a relation with `schema` and `database` path keys intact. 
- `postgres__make_temp_relation`, `postgres__make_intermediate_relation`, and `postgres__make_backup_relation` all call a `postgres__make_relation_with_suffix` macro to avoid repeated code and then return the relation for their respective uses. 
- Added bonus of not removing `postgres__make_temp_relation` & alleviating your breaking changes consideration
- In the global project, `default__make_intermediate_relation` calls `default__make_temp_relation` since the non-temporary schema consideration does not apply on other data warehouses (as far as I understand at least - please let me know if this is incorrect).

Finally, I have a question that may affect the utility of `postgres__make_relation_with_suffix`, but they're more appropriate for [the testing thread](https://github.com/dbt-labs/dbt-core/pull/4921#discussion_r833019263).